File: packages/android/fernlink-sdk/src/main/kotlin/xyz/fernlink/sdk/ble/BleUuids.kt
package xyz.fernlink.sdk.ble

import java.util.UUID

object BleUuids {
    val FERNLINK_SERVICE:   UUID = UUID.fromString("fern0000-0000-1000-8000-00805f9b34fb")
    val CHAR_REQUEST:       UUID = UUID.fromString("fern0001-0000-1000-8000-00805f9b34fb")
    val CHAR_PROOF:         UUID = UUID.fromString("fern0002-0000-1000-8000-00805f9b34fb")
    val CHAR_STATUS:        UUID = UUID.fromString("fern0003-0000-1000-8000-00805f9b34fb")
    val DESCRIPTOR_CCC:     UUID = UUID.fromString("00002902-0000-1000-8000-00805f9b34fb")

    const val MTU = 512
}

File: packages/ios/Sources/FernlinkSDK/BleUuids.swift
import CoreBluetooth

// fe4e = valid hex encoding of "FeN" (Fernlink). Must match Android, Rust, and TypeScript layers.
enum BleUuids {
    static let fernlinkService = CBUUID(string: "FE4E0000-0000-1000-8000-00805F9B34FB")
    static let charRequest     = CBUUID(string: "FE4E0001-0000-1000-8000-00805F9B34FB")
    static let charProof       = CBUUID(string: "FE4E0002-0000-1000-8000-00805F9B34FB")
    static let charStatus      = CBUUID(string: "FE4E0003-0000-1000-8000-00805F9B34FB")
    static let descriptorCCC   = CBUUID(string: "2902")

    static let mtu = 512
}
